content/en/tracing/trace_ingestion/mechanisms.md

```diff
@@ -0,0 +1,193 @@
+
+The agent's target traces-per-second can be set in its main configuration file (`datadog.yaml`) or as an environment variable :
+```
+@param max_traces_per_second - integer - optional - default: 10
+@env DD_APM_CONFIG_MAX_TRACES_PER_SECOND - integer - optional - default: 10
+```
+
+### In Tracing librairies : User-defined Rules
+_ingestion\_reason : rule_
+
+At the library level, more specific sampling configuration is available:
+- Set a specific sampling rate to apply to all root services, overriding the agent's [default mechanism](#default-mechanism).
+- Set a sampling rate for a specific root service.
+- Set a limit on the number of ingested traces-per-second.
+
+**Note :** These rules also follow a head-based sampling mechanism. If the traffic for a service is higher than the configured maximum traces-per-second, then traces are dropped at the root. It does not create incomplete traces.
+
+The configuration can be set directly in the code, or as environment variables :
+
+{{< tabs >}}
+{{% tab "Environment variables" %}}
+
+```
+@env  DD_TRACE_SAMPLE_RATE - integer - optional null (defaults to agent default feedback loop)
+@env DD_TRACE_SAMPLING_RULES - integer - optional null
+@env  DD_TRACE_RATE_LIMIT - integer - optional 100 (if using the agent default mechanism, the rate limiter is ignored)
+```
+
+{{% /tab %}}
+{{% tab "Code API" %}}
+
+```
+# in dd-trace-py
+# Sample 10% of all traces with rate limit of 100 traces per second sampled
+# and an override sample rate for a specific service
+tracer.configure(sampler=DatadogSampler(
+    default_sample_rate=0.10, # keep 10% of traces
+    rate_limit=100, # but at most 100 traces per second
+    rules=[
+      # 100% sampled for “my-service”, but the 100 traces-per-second overall limit is still honored
+      SamplingRule(sample_rate=1.0, service=’my-service’),
+    ],
+)
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+Read more about configuring the ingestion in the [tracing librairies][1] documentation
+
+**Note :** Services configured with user-defined rules are marked as `Configured` in the [Ingestion Control Page][3] Configuration column. Services configured to use the default mechanism are labeled as `Automatic`.
+
+## Force Keep and Drop
+_ingestion\_reason : manual_
+
+Head-based sampling mechanism can be overriden at the tracing library level. For instance, if you need to monitor a critical transaction, force the associated trace to be kept. On the other hand, for unnecessary or repetitive information like health checks, force the trace to be dropped.
+
+- Set `ManualKeep` on a span to indicate that it and all child spans should be ingested. The resulting trace might appear incomplete in the UI if the span in question is not the root span of the trace.
+```
+// in dd-trace-go
+span.SetTag(ext.ManualKeep, true)
+```
+- Set ManualDrop to make sure that **no** child span is ingested. [Error and rare samplers](#error-and-rare-traces) will be ignored in the agent.
```

Comment:
```suggestion
- Set ManualDrop to make sure that **no** child span, not even [error and rare traces](#error-and-rare-traces), is ingested.
```

Reply:
These are 2 different mechanisms (one in the tracing libraries, on in the agent. We need to make the clear distinction in the sentence